govcd/vdc.go

```diff
@@ -837,3 +836,41 @@ func (vdc *Vdc) buildNsxvNetworkServiceEndpointURL(optionalSuffix string) (strin
 
 	return hostname, nil
 }
+
+// GetEdgeGatewayReferenceList retrieves a list of edge gateway references.
+// Unlike other entities under Vdc, edge gateway are not listed in the parent entities
+// and therefore we need to get the list using a query.
+func (vdc *Vdc) GetEdgeGatewayReferenceList(refresh bool) ([]*types.Reference, error) {
+
+	if refresh {
+		err := vdc.Refresh()
+		if err != nil {
+			return nil, fmt.Errorf("error refreshing vdc: %s", err)
+		}
+	}
+	for _, av := range vdc.Vdc.Link {
+		if av.Rel == "edgeGateways" && av.Type == "application/vnd.vmware.vcloud.query.records+xml" {
```

Comment:
Could potentially be a constant as we have it for some other types. Not a necessity.

Reply:
In fact, we had a constant, although it contained a different text and was unused.
Now I put it to good use with the right contents.